valkyrie/lib/valkyrie/persistence/postgres/resource_factory.rb

```diff
@@ -14,8 +14,15 @@ def to_model(orm_object)
       # @return [Valkyrie::Persistence::Postgres::ORM::Resource] ActiveRecord
       #   model for the Valkyrie model.
       def from_model(resource)
-        ::Valkyrie::Persistence::Postgres::ORM::Resource.find_or_initialize_by(id: resource.id.to_s).tap do |orm_object|
+        if resource.created_at.blank? && resource.id.present?
```

Comment:
Same semantic check... Thinking more that we have a half-and-half responsibility to model here.